lib/src/header/all_day_events.dart

```diff
@@ -42,15 +44,25 @@ class AllDayEvents<E extends Event> extends StatelessWidget {
               events = events.where((e) => e.intersectsInterval(visibleDates));
 
               return ValueListenableBuilder(
-                valueListenable: controller.scrollControllers.pageListenable,
-                builder: (context, page, __) =>
-                    _buildEventLayout(context, events, page),
-              );
+                  valueListenable: controller.scrollControllers.pageListenable,
+                  builder: (context, page, __) => GestureDetector(
+                        behavior: HitTestBehavior.translucent,
+                        onTapUp: (details) {
+                          final tappedCell = details.localPosition.dx / (constraints.widthConstraints().maxWidth / controller.visibleRange.visibleDays);
+                          final date = LocalDate.fromEpochDay(page.floor() + tappedCell.toInt());
+                          final dateAndTime = DateTime(date.year, date.monthOfYear, date.dayOfYear + 2);
+                          final startTime = LocalDateTime.dateTime(dateAndTime);
+                          if (onCreateAllDayEvent != null) {
+                            onCreateAllDayEvent(startTime, true);
+                          }
```

Comment:
Similar to the part-day code, I'd prefer extracting this to a private helper method and only conditionally assigning it to `onTapUp` if we're actually interested in the tap.